### PR TITLE
A way to override base_url (#802)

### PR DESCRIPTION
**Pull request by xeor**
Fixes https://github.com/music-assistant/hass-music-assistant/issues/802

### CODE REVIEW

File: music_assistant/controllers/streams.py [hunk 1]
```diff
@@ -56,6 +57,18 @@ def __init__(self, mass: MusicAssistant):
     @property
     def base_url(self) -> str:
         """Return the base url for the stream engine."""
+
+        """
+        This is a purpously undocumented feature to override the automatic
+        generated base_url used by the streaming-devices.
+
+        If you need this, you know it, but you should probably try to not set it!
```
Comment (marcelveldt) [inline music_assistant/controllers/streams.py:65]: haha!

Comment (marcelveldt) [reply music_assistant/controllers/streams.py:65]: Can you change the comment just to be inline and leave the function docstring alone ?

File: music_assistant/controllers/streams.py [hunk 2]
```diff
@@ -56,6 +57,18 @@ def __init__(self, mass: MusicAssistant):
...
+        Also see https://github.com/music-assistant/hass-music-assistant/issues/802
+        and https://github.com/music-assistant/hass-music-assistant/discussions/794#discussioncomment-3331209
+        """
+        if BASE_URL_OVERRIDE_ENVNAME in os.environ:
+            return os.environ[BASE_URL_OVERRIDE_ENVNAME]
```
Comment (marcelveldt) [inline music_assistant/controllers/streams.py:70]: insert in-line comment here (so appended with #)

File: music_assistant/controllers/streams.py (hunk 1 above)
Comment (xeor) [reply music_assistant/controllers/streams.py:65]: Just the way you wanted it... ;D

File: music_assistant/controllers/streams.py (hunk 2 above)
Comment (xeor) [reply music_assistant/controllers/streams.py:70]: You mean changing the whole """ style comment to # style?

Comment (marcelveldt) [reply music_assistant/controllers/streams.py:70]: correct and put it between line 69 and 70 to make it clear it belongs to that line

File: music_assistant/controllers/streams.py [hunk 3]
```diff
@@ -56,6 +57,16 @@ def __init__(self, mass: MusicAssistant):
     @property
     def base_url(self) -> str:
         """Return the base url for the stream engine."""
+
+        if BASE_URL_OVERRIDE_ENVNAME in os.environ:
+            # This is a purpously undocumented feature to override the automatic
+            # generated base_url used by the streaming-devices.
+            #
```
Comment (marcelveldt) [inline music_assistant/controllers/streams.py:64]:
```suggestion
```

File: music_assistant/controllers/streams.py [hunk 4]
```diff
@@ -11,7 +12,7 @@
 
 from aiohttp import web
 
-from music_assistant.constants import FALLBACK_DURATION, SILENCE_FILE
+from music_assistant.constants import FALLBACK_DURATION, SILENCE_FILE, BASE_URL_OVERRIDE_ENVNAME
```
Comment (marcelveldt) [inline music_assistant/controllers/streams.py:15]:
```suggestion
from music_assistant.constants import (
    FALLBACK_DURATION, SILENCE_FILE, BASE_URL_OVERRIDE_ENVNAME
)
```

Comment (marcelveldt) [reply music_assistant/controllers/streams.py:15]: otherwise the linter fails due to max line length

Comment (xeor) [reply music_assistant/controllers/streams.py]: Yea, I just noticed.. Didn't think about that :) thanks

Comment (marcelveldt) [reply music_assistant/controllers/streams.py]: CI is still failing. Hold on I'll quickly run the linter against your code.

**Comment by marcelveldt** [comment]
Thanks for your understanding and your PR!
Enjoy your music!

**Pull request merged**